### PR TITLE
Added CLI tooling for querying block/transaction/account details

### PR DESCRIPTION
**Pull request by AionJayT**
## Notice

It is not allowed to submit your PR to the 'master' branch directly, please submit your PR to the 'master-pre-merge' branch and rebase your PR to 'master-pre-merge' branch.

## Description

Please include a brief summary of the change that this pull request proposes. Include any relevant motivation and context. List any dependencies required for this change.

- Added new CLI tooling for query block/transaction/account details.
- To query the account details currently only has the API, does not have the implement details because it relates with the database account retrieve model revise.

Fixes Issue # .

## Type of change

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [ ] Bug fix.
- [ ] New feature.
- [x] Enhancement.
- [ ] Unit test.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

## Testing

Please describe the tests you used to validate this pull request. Provide any relevant details for test configurations as well as any instructions to reproduce these results.

- A test case for the CLI task priorities.

### CODE REVIEW

File: modAionImpl/src/org/aion/zero/impl/cli/Arguments.java [hunk 1]
```diff
@@ -167,6 +167,28 @@
                     "drops all databases except for block and index when not given a parameter or starting from 0 and redoes import of all known main chain blocks")
     private String redoImport = null;
 
+    // data query
+    @Option(
+        names = {"qb", "-q block", "--query block"},
```
Comment (AlexandraRoatis) [inline modAionImpl/src/org/aion/zero/impl/cli/Arguments.java]: Parameters with spaces are problematic for the CLI library. We keep the old ones for legacy reasons, but I would recommend not introducing more.

I suggest changing the options to `names = {"qb", "--query-block"}`.
Similarly below: `names = {"qt", "--query-tx"}` and `names = {"qa", "--query-account"}`.

File: modAionImpl/src/org/aion/zero/impl/cli/Arguments.java [hunk 2]
```diff
@@ -167,6 +167,28 @@
...
+        arity = "1",
+        paramLabel = "<block_number>",
+        description = "retrieve block information")
+    private String blockDetails = null;
+
+    @Option(
+        names = {"qt", "-q tx", "--query tx"},
+        arity = "1",
+        paramLabel = "<transaction hash>",
```
Comment (AlexandraRoatis) [inline modAionImpl/src/org/aion/zero/impl/cli/Arguments.java]: Inconsistent `paramLabel` format. It should be `<transaction_hash>` and `<account_address>`.

File: modAionImpl/src/org/aion/zero/impl/cli/Cli.java [hunk 3]
```diff
@@ -69,6 +72,9 @@ public int getValue() {
         VERSION,
         CONFIG,
         INFO,
+        QUERY_BLOCK,
```
Comment (AlexandraRoatis) [inline modAionImpl/src/org/aion/zero/impl/cli/Cli.java]: This ordering does not reflect the intended priority. For consistency, the location should match the ordering in both `call`, `getBreakingTaskPriority` and `getSkippedTasks`.

File: modAionImpl/src/org/aion/zero/impl/db/DBUtils.java [hunk 4]
```diff
@@ -703,4 +706,117 @@ public AionBlockSummary getRight() {
 
         System.out.println("Importing stored blocks COMPLETE.");
     }
+
+    /**
+     * @implNote Used by the CLI call.
+     */
+    public static Status queryTransaction(byte[] txHash) {
```
Comment (AlexandraRoatis) [inline modAionImpl/src/org/aion/zero/impl/db/DBUtils.java]: Please add some unit tests to cover these methods.

File: modAionImpl/src/org/aion/zero/impl/cli/Cli.java (hunk 3 above)
Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/cli/Cli.java]: revised.

File: modAionImpl/src/org/aion/zero/impl/cli/Arguments.java (hunk 2 above)
Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/cli/Arguments.java]: revised.

File: modAionImpl/src/org/aion/zero/impl/db/DBUtils.java (hunk 4 above)
Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/db/DBUtils.java]: Added the tests for covering the methods.

File: modAionImpl/src/org/aion/zero/impl/AionBlockchainImpl.java [hunk 5]
```diff
@@ -1135,6 +1135,11 @@ public AionBlockSummary add(AionBlock block, boolean rebuild) {
         return summary;
     }
 
+    public AionBlockSummary addNoFlush(AionBlock block) {
```
[the remark this answers is not in the document]
Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/AionBlockchainImpl.java]: updated

**Pull request merged**